File: app/src/main/java/me/madsbf/launcher/model/MainSwatch.java
package me.madsbf.launcher.model;

import android.graphics.Color;
import android.support.v7.graphics.Palette;

public class MainSwatch {

    int colorPrimary;
    int colorPrimaryDark;
    int colorAccent;

    public MainSwatch(Palette.Swatch swatch) {
        colorPrimary = swatch.getRgb();

        float[] hsv = new float[3];
        Color.colorToHSV(colorPrimary, hsv);
        hsv[2] *= 0.8f;
        colorPrimaryDark = Color.HSVToColor(hsv);

        // TODO
        colorAccent = colorPrimary;
    }

    public int getColorPrimary() {
        return colorPrimary;
    }

    public int getColorPrimaryDark() {
        return colorPrimaryDark;
    }

    public int getColorAccent() {
        return colorAccent;
    }
}

File: app/src/main/kotlin/me/madsbf/launcher/model/MainSwatch.kt
package me.madsbf.launcher.model

import android.graphics.Color
import android.support.v4.graphics.ColorUtils
import android.support.v7.graphics.Palette
import me.madsbf.launcher.view.utils.PaletteUtils

public class MainSwatch(palette: Palette) {

    public val colorPrimary: Int
    public val colorPrimaryDark: Int
    public val colorAccent: Int

    init {
        val swatch = PaletteUtils.getMostVibrantSwatch(palette)
        colorPrimary = swatch.rgb

        val hsv = FloatArray(3)
        Color.colorToHSV(colorPrimary, hsv)
        hsv[2] *= 0.8f
        colorPrimaryDark = Color.HSVToColor(hsv)

        // TODO
        colorAccent = colorPrimary
    }
}
